src/main/java/com/chikli/hudson/plugin/naginator/NaginatorRetryAction.java

```diff
@@ -73,12 +77,27 @@ static boolean scheduleBuild(final AbstractBuild<?, ?> build, final int delay, i
     }
 
     static boolean scheduleBuild(final AbstractBuild<?, ?> build, final int delay, final NaginatorAction action) {
+        return scheduleBuild2(build, delay, action, false);
+    }
+
+    static boolean scheduleBuild2(final AbstractBuild<?, ?> build, final int delay, final NaginatorAction action, boolean replaceUser) {
         final List<Action> actions = new ArrayList<>();
+        NaginatorCause cause = new NaginatorCause(build);
         actions.add(action);
+        action.setCause(cause);
         actions.add(build.getAction(ParametersAction.class));
-        actions.add(build.getAction(CauseAction.class));
+        actions.add(getCauseAction(build, replaceUser, cause));
 
-        return build.getProject().scheduleBuild(delay, new NaginatorCause(build), actions.toArray(new Action[0]));
+        return build.getProject().scheduleBuild2(delay, actions.toArray(new Action[0])) != null;
     }
 
+    private static CauseAction getCauseAction(AbstractBuild<?, ?> build, boolean replaceUser, NaginatorCause cause) {
+        List<Cause> currentCauses = new ArrayList<>(build.getCauses());
+        List<Cause> newCauses = currentCauses.stream().filter(c -> !((c instanceof Cause.UserIdCause) && replaceUser) && !(c instanceof NaginatorCause)).collect(Collectors.toList());
```

Comment:
Please add some more tests to improve coverage here.

Reply:
done